realm/realm-library/src/androidTest/kotlin/io/realm/UUIDTests.kt

```diff
@@ -0,0 +1,590 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.annotations.PrimaryKey
+import io.realm.annotations.Required
+import io.realm.exceptions.RealmException
+import io.realm.exceptions.RealmPrimaryKeyConstraintException
+import io.realm.kotlin.createObject
+import io.realm.kotlin.where
+import io.realm.rule.TestRealmConfigurationFactory
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+import kotlin.test.assertFailsWith
+
+open class UUIDPrimaryKeyRequired
+    : RealmObject() {
+    @PrimaryKey
+    @Required
+    var id: UUID? = null
+    var name: String = ""
+    var anotherId: UUID? = null
+}
+
+open class UUIDPrimaryKeyNotRequired
+    : RealmObject() {
+    @PrimaryKey
+    var id: UUID? = null
+    var name: String = ""
+}
+
+open class UUIDAndString
+    : RealmObject() {
+    var id: UUID? = null
+    var name: String = ""
+}
+
+open class UUIDRequiredRealmList
+    : RealmObject() {
+    var id: Long = 0
+
+    @Required
+    var ids: RealmList<UUID> = RealmList()
+    var name: String = ""
+}
+
+open class UUIDOptionalRealmList
+    : RealmObject() {
+    var id: Long = 0
+
+    var ids: RealmList<UUID> = RealmList()
+    var name: String = ""
+}
+
+@RunWith(AndroidJUnit4::class)
+class UUIDTests {
```

Comment:
Nice test coverage 👍 

Two things I see missing is:
- Test for linked queries, i.e. `equalTo("parentObj.uuidField", uuid)`
- RealmObjectSchema tests, i.e. `RealmObjectSchema.getFieldType() == RealmFieldType.UUID`